src/Config.php

```diff
@@ -62,11 +44,41 @@ public function __construct($config)
 
     /**
      * PDO connect options
-     *
      * Array of key-value pairs that should be set when calling {@link Connection::connect()} in order to establish a DB
      * connection.
      *
      * @var array
      */
     public $options;
+
+    /**
+     * Create a new SQL connection configuration from the given configuration key-value pairs
+     *
+     * The given configuration is constraint to the class public properties.
+     * Keys will be converted to camelCase, e.g. use_ssl → useSsl.
+     *
+     * @param iterable $config Configuration key-value pairs
+     *
+     * @throws InvalidArgumentException If $config is not iterable
+     */
+    public function __construct($config)
+    {
+        if (! is_iterable($config)) {
+            throw new InvalidArgumentException(sprintf(
+                '%s expects parameter one to be iterable, got %s instead',
+                __METHOD__,
+                get_php_type($config)
+            ));
+        }
+
+        foreach ($config as $key => $value) {
+            $camelKey = Str::camel($key);
+
+            if (!property_exists(static::class, $camelKey)) {
```

Comment:
Space missing after `!`.

Reply:
Can you please check whether adding `Generic.Formatting.SpaceAfterNot` to our phpcs.xml would help to let our CI fail in such cases?